server/server.go

```diff
@@ -917,11 +917,14 @@ func (s *Server) setupGRPC() error {
 		return err
 	}
 
-	go func() {
+	// Start server with infinite retries
+	go common.RetryForever(context.Background(), time.Second, func(context.Context) error {
```

Comment:
What if parameters for staring servers are wrong? like port is already used or similar

Reply:
if a port is in use, the app will self heal on retry when the port is made free. this is desired behaviour, instead of continuing a process that is broken (e.g. not relaying or handling deposits)

Reply:
I don't find this a good practice, infinitely waiting for the port to be free is not a desired behaviour. The port can be busy forever.

Reply:
Can refactor this to return error from parent func

Reply:
Happy to rm this code of course.

Agreed that a port could be busy forever. It would become freed by an operator. At that point, Edge would "self heal" with respect to this functionality, which is a handy property for a distributed app.

For context, we run on Kubernetes so a port clash will never happen.

Reply:
FWIW, in the past I've occasionally applied a 'warn + sleep N times and then fail' for some of these type of instances where failure could be transient but also catastrophic if not resolved. A port being free might be an example where it might be expected to be transiently occupied - i.e. while something else shuts down - so it may be useful to retry a few times. But then if after some time - say 10s of trying - it doesn't resolved then fail.
But I recognize that it's not a good idea to blankety apply this idea so it may need to be done on a case-by-case basis.

Reply:
Yep and if we do decide to employ retries, we have the flexibility to customize the retry logic based on errors returned from the retryable functionality. RetryForever() is a simple/dumb approach that can be used as stop-gap measure